src/services/tx/txMonitor.ts

```diff
@@ -33,3 +36,52 @@ export const waitForTx = async (provider: JsonRpcProvider, txId: string, txHash:
     })
   }
 }
+
+export const waitForRelayedTx = (taskId: string, txId: string): void => {
+  const gelato = new GelatoRelay()
+
+  const checkTxStatus = async () => {
+    const taskStatus = await gelato.getTaskStatus(taskId)
+
+    switch (taskStatus?.taskState) {
+      case 'CheckPending':
+      case 'ExecPending':
+      case 'WaitingForConfirmation':
```

Comment:
Can we create an enum for the `taskState`s?

Reply:
I will have to use
```
switch (taskStatus.taskState as string) {
```

as we get `taskStatues.taskState` typed from the Gelato request and they don't export that type. Should I create the enum anyway?

Reply:
That sounds good. In general everything that concerns the Gelato SDK would be nice to be consumed from our `relaying-service` if possible

Reply:
I think I misread this comment the other day as I was trying to type `taskState`. I've created the Enum for the TaskStates.